### PR TITLE
🐛  Fix github sbt-ci-release plugin generated workflow by using `crossScalaVersions` and removing ++ from build and test job

### PR DESCRIPTION
**Pull request by jeffmay**
## Description

- Use `githubWorkflowScalaVersions` to select all Scala versions
- Attempt to use `crossScalaVersions` with to generate a `ci.yml` file that runs the correct scala versions on subprojects with matrix

~~For some reason, this doesn't work. It appears the `++` doesn't work as [documented](https://www.scala-sbt.org/1.x/docs/Cross-Build.html#Switching+Scala+version).~~

UPDATE: I was able to get this to generate the correct Scala version by setting `ThisBuild / scalaVersion`. This workflow should now work.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by mdedetrich** [comment]
Is this also related to https://github.com/rallyhealth/scalacheck-ops/issues/60?

**Comment by jeffmay** [comment]
> 

This PR only exists for reference of [this issue](https://github.com/djspiewak/sbt-github-actions/issues/118)

File: build.sbt [hunk 1]
```diff
@@ -28,6 +31,31 @@ mimaFailOnNoPrevious := false
...
+          "PGP_SECRET" -> "${{ secrets.PGP_SECRET }}",
+          "SONATYPE_PASSWORD" -> "${{ secrets.SONATYPE_PASSWORD }}",
+          "SONATYPE_USERNAME" -> "${{ secrets.SONATYPE_USERNAME }}"
+        )
+      )
+    ),
+    // Remove the ++ from the preambles since we use sbt-projectmatrix
+    githubWorkflowBuildSbtStepPreamble := Seq(),
+    // githubWorkflowPublishSbtStepPreamble := Seq() // Doesn't compile
+    githubWorkflowPublishPreamble := Seq() // Doesn't do anything
```
Comment (jeffmay) [inline build.sbt]: Remove

**Review by mingy711 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,66 +23,78 @@ jobs:
...
+
+      - name: Check that workflows are up to date
+        run: sbt githubWorkflowCheck
 
       - name: Build project
         run: sbt test
 
+      - name: Make target directories
+        if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/v'))
+        run: mkdir -p core/target/1-15-jvm-2.13 core/target/1-js-3 core/target/1-js-2.13 joda/target/1-13-jvm-2.12 core/target/1-js-2.12 core/target/1-13-jvm-2.12 core/target/1-15-jvm-2.12 core/target/1-15-js-2.12 joda/target/1-15-jvm-3 joda/target/1-15-jvm-2.13 core/target/1-15-js-2.13 core/target/1-jvm-2.12 joda/target/1-14-jvm-2.13 core/target/1-native-3 target core/target/1-15-jvm-3 core/target/1-15-js-3 joda/target/1-14-jvm-2.12 core/target/1-15-native-3 joda/target/1-jvm-2.13 joda/target/1-15-jvm-2.12 core/target/1-14-jvm-2.12 joda/target/1-jvm-2.12 core/target/1-14-jvm-2.13 core/target/1-jvm-2.13 core/target/1-jvm-3 joda/target/1-jvm-3 core/target/1-15-native-2.13 core/target/1-native-2.13 project/target
```
Comment (paul-a-kennedy-rally) [inline .github/workflows/ci.yml:87]: I'm guessing that extra directories would not kill this.  If so, you could simplify this greatly:

```suggestion
        run: mkdir -p {core,joda}/target/1-{13,14,15,}-{js,jvm,native}-{2.12,2.13,3} {project/,}target
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -16,66 +23,78 @@ jobs:
...
-          joda/target/1-15-jvm-2.12 \
-          joda/target/1-15-jvm-2.13 \
-          joda/target/1-15-jvm-3 \
-          joda/target/1-jvm-2.12 \
-          joda/target/1-jvm-2.13 \
-          joda/target/1-jvm-3 \
-          project/target \
-          target
+        if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/v'))
+        run: tar cf targets.tar core/target/1-15-jvm-2.13 core/target/1-js-3 core/target/1-js-2.13 joda/target/1-13-jvm-2.12 core/target/1-js-2.12 core/target/1-13-jvm-2.12 core/target/1-15-jvm-2.12 core/target/1-15-js-2.12 joda/target/1-15-jvm-3 joda/target/1-15-jvm-2.13 core/target/1-15-js-2.13 core/target/1-jvm-2.12 joda/target/1-14-jvm-2.13 core/target/1-native-3 target core/target/1-15-jvm-3 core/target/1-15-js-3 joda/target/1-14-jvm-2.12 core/target/1-15-native-3 joda/target/1-jvm-2.13 joda/target/1-15-jvm-2.12 core/target/1-14-jvm-2.12 joda/target/1-jvm-2.12 core/target/1-14-jvm-2.13 core/target/1-jvm-2.13 core/target/1-jvm-3 joda/target/1-jvm-3 core/target/1-15-native-2.13 core/target/1-native-2.13 project/target
```
Comment (paul-a-kennedy-rally) [inline .github/workflows/ci.yml:91]: I think that it would be better _not_ to collapse this on one line.

**Review by paul-a-kennedy-rally — APPROVED** [review]
I think that you could greatly improve readability by using the shell brace expansion.

Barring that, I think that the one element per line is much more readable (and easier to automate tooling to process as well).

Not egregious enough as to block this PR.

**Pull request merged**